main/verifier.go

```diff
@@ -135,54 +62,113 @@ func loadUPP(hashString string, conf Config) ([]byte, int, error) {
 			}
 		}
 	}
+	//noinspection GoUnhandledErrorResult
+	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, resp.StatusCode, fmt.Errorf("could not (yet) retrieve certificate for hash %s from verification service (%s): %s", hashString, verificationURL, resp.Status)
+		return nil, resp.StatusCode, fmt.Errorf("could not (yet) retrieve certificate for hash %s from verification service (%s): %s", hashString, verifyService, resp.Status)
 	}
 
 	vf := verification{}
 	err = json.NewDecoder(resp.Body).Decode(&vf)
 	if err != nil {
 		return nil, http.StatusInternalServerError, fmt.Errorf("unable to decode verification response: %v", err)
 	}
-	_ = resp.Body.Close()
-	return vf.UPP, resp.StatusCode, nil
+	return vf.UPP, http.StatusOK, nil
+}
+
+// verifyUPP verifies the validity of a UPP's signature. Requests public key from the key service if unknown.
+func verifyUPP(p *ExtendedProtocol, upp []byte, keyService string) (uuid.UUID, error) {
+	o, err := ubirch.DecodeChained(upp)
```

Comment:
do we only decode "chained UPPs" What if I want to get verify a signed UPP?

Reply:
yes, we currently only support the verification of chained UPPs with the client. this method shall be extended soon.